### DISCUSSION TITLE
Question about audiokit-sd_wav

### DISCUSSION THREAD

**Discussion by richyxi** [pschatzmann/arduino-audio-tools#988] [Q&A] [answered]
Hi pschatzmann! Hope you doing great!

I have a couple of questions regarding saving sound on a file. I been trying to build an application that hears music and saves it on a file every 10-15 seconds so i look to the documentation i got this.

first all the instances:
```
AudioInfo info(16000, 1, 16);
AudioKitStream in;
File file;  // final output stream
EncodedAudioStream out(&file, new WAVEncoder());
StreamCopy copier(out, in); // copies data
```

and then, the loop:

```
void loop() {


    Serial.println("Recording...");
    copier.copyMs(10000, info);
    Serial.println("Finish recording...");  
    
    file.flush();
    Serial.print("File has ");
    Serial.print(file.size());
    Serial.println(" bytes");
    Serial.println("Wait");
    delay(5000);
    
  }
```

but what i found is that in each iteration the recorded sound will be appended, and i need the file with just the last 15 seconds recorded.

Is there a way to make an overwrite operation each time  `copier.copyMs(10000, info);` is executed. I been diving on the docs and seems that there is a way to call seek() which i think it specified a position in file to maybe start the write(?) but im kinda lost here.

Thank you in advance and hope you had a great day!

**Comment by pschatzmann** [comment, marked as answer]
In the Wiki https://github.com/pschatzmann/arduino-audio-tools/wiki/Encoding-and-Decoding-of-Audio you find:

- Before you start to write to a file, delete the file or move to the beginning. Otherwise the content is just appended!

This has nothing to do with my project: here is the link to the official SD documentation: https://www.arduino.cc/reference/en/libraries/sd/
I suggest that you have a look at the seek method!

**Reply by richyxi** [reply]
Thanks for the quick response, i really appreciate it!!

I tried the seek() method like this..

```
void loop() {


    file.seek(0);
    Serial.println("Recording...");
    copier.copyMs(10000, info);
    Serial.println("Finish recording...");  
    file.flush();
    
    Serial.print("File has ");
    Serial.print(file.size());
    Serial.println(" bytes");
    Serial.println("Wait");
    delay(5000);
    
  }
```


 and i got the following:

1. The information is not longer being appended.
2. When i try to listen to the file just the first iteration is saved, when the info is overwritted it seems that the data is corrupted.


```
01:44:33.503 -> Recording...
01:44:43.462 -> Finish recording...   // I can hear the audio writed here
01:44:43.495 -> File has 319532 bytes
01:44:43.495 -> Wait

01:44:48.479 -> Recording...
01:44:58.293 -> Finish recording...   // When i try to hear the second iteration from the sd card is when faced the issue
01:44:58.293 -> File has 319532 bytes
01:44:58.293 -> Wait
```

Indeed this makes an overwrite but its pretty interesting that only works when the data is written for the first time.
Do you think im missing something here? Any pointer will be much appreciated.

Thanks in advance!

**Comment by pschatzmann** [comment]
I would open the file in the beginning of the loop and at the end of the loop close it again.

**Reply by richyxi** [reply]
I check different files and it seems that just after the first file write, the remaining ones had no header written. I build a script on python that manually puts the missing parts and I was able to listen those.

This counts as a bug or maybe I’m missing something in my sketch? 

I can help with some testing 👍

**Comment by pschatzmann** [comment]
No it's not: it's doing exectly what you tell it to do!

I don't like to answer to questions which are already properly documented in the Wiki. Please re-read the chapter about Encoding carefully again!
